packages/contract/lib/execute.js

```diff
@@ -56,7 +57,23 @@ const execute = {
               : accept(bestEstimate.toHexString());
           }
         })
-        .catch(() => accept());
+        .catch(error => {
```

Comment:
Yeah -- I agree that this way of doing things is pretty wonky.  I don't really think it makes a lot of sense that `stacktrace` is a parameter to `interfaceAdapter.estimateGas`; IMO it would make sense to have the catch here, not there.  But since it already is that way, I decided not to do extra work by changing it.  However, I can totally change it if you think that would be a good idea.

Reply:
Well, no, it's not a matter of what error it would throw; it's a matter of whether stacktracing is on or not.  My point is about where that should be checked; to my mind, that should be checked here, not in `estimateGas`.

Reply:
Yes that is what I was trying to say. But I'm also saying that we need to check in the `catch` to make sure it is throwing for the right reason...unless the only case in which `estimateGas` fails is when something reverts.

So if something reverts, `estimateGas` will throw, right? In the case that stacktracing is on, we can ignore the error and proceed. Will `estimateGas` ever fail for a different reason? In that case we would want to throw and not continue regardless of whether stacktracing is on. Does that seem correct?

Reply:
Oh, that's a good point, I didn't think of that.  Yeah I agree there's a potential problem there.  Perhaps that can be saved for a later PR, though, since the problem is pre-existing?  Detecting the error type may require some trickiness in general.

Reply:
Sounds good to me, I'll leave it up to you!

Reply:
Cool, I'll leave it for a later PR, and file an issue once this is merged.